### ISSUE TITLE
Stuck User - Choice.trim is not a function thrown in done.js

### ISSUE THREAD

**Issue by alonsonic** [Microsoft/BotBuilder#2408]
Hi guys, I'm having an issue with my bot. 

An important user account is stuck in a loop, we made a change to the choice prompt and added an invalid third parameter (not an string array), this is fixed now but any message sent by the user throws an error.

I saw similar issues already posted but none have a clear answer on how to unstuck the affected users.

What should I do? I'm running the latest version of the bot connector in a Node.js server.

**Comment by alonsonic** [comment]
I got this error just before the user got stuck. Maybe he is waiting for a choice in the list but there is no valid choice to pick.

{"error":{"message":"(#100) Param message[quick_replies][0][title] must be a UTF-8 encoded string","type":"OAuthException","code":100,"fbtrace_id":"FxA1o2CWH5u"}}

**Comment by Stevenic** [comment]
Do you have a `bot.endConversationAction()`? If you add this to your bot they can just say "goodbye" to reset their state and start over.  The second option is you can add a dialogVersion() middleware:

```
bot.use(builder.Middleware..dialogVersion({ version: 1.0 }));
```

Anyytime you make a major change to your bot you can increment this version number and force to have all current conversations ended.

**Comment by josephktcheung** [comment]
@alonsonic I too had this problem today. I solved it by clearing all the userData + conversationData for that particular user.

I tried `bot.endConversationAction()` but didn't work. Didn't try dialogVersion() middleware. Removing userData + conversationData works for me but it should probably be the last resort if solutions suggested by @Stevenic don't work.

What I did:

1. Get an authorization token as per MS botframework REST API for [authentication](https://docs.botframework.com/en-us/restapi/authentication#bot---connector-step-1-post-to-the-msaaad-v2-login-service-to-retrieve-a-token)
2. Submit a DELETE request with the authorization token in the header to `https://state.botframework.com/v3/state/{channelId e.g. facebook}/users/{userId}` as per documentation [here](https://docs.botframework.com/en-us/restapi/state/#!/BotState/BotState_DeleteStateForUser)

This should clean up all conversationData + userData, i.e. user is back to a clean state.
